File: Client/src/models/studyNote.ts
import type { ContentAccessLevel, ReviewStatus } from './enums'

export interface StudyNote {
  id: string
  topicId: string
  title: string
  content: string
  sourceReference?: string | null
  isAiGenerated: boolean
  reviewStatus: ReviewStatus
  accessLevel: ContentAccessLevel
  reviewedBy?: string | null
  reviewedAt?: string | null
  reviewComment?: string | null
}

export interface PublicStudyNote {
  id: string
  topicId: string
  topicTitle: string
  courseId: string
  courseName: string
  licenseId: string
  licenseName: string
  title: string
  content: string
  sourceReference?: string | null
}

export interface CreateStudyNote {
  topicId: string
  title: string
  content: string
  sourceReference?: string | null
  isAiGenerated: boolean
  reviewStatus: ReviewStatus
  accessLevel: ContentAccessLevel
}

export type UpdateStudyNote = CreateStudyNote

File: API/Entities/StudyNote.cs
namespace API.Entities;

public class StudyNote : ModeratedEntity
{
    public Guid TopicId { get; set; }

    public Topic? Topic { get; set; }

    public string Title { get; set; } = string.Empty;

    public string Content { get; set; } = string.Empty;

    public string? SourceReference { get; set; }

    public bool IsAiGenerated { get; set; }
}
